### PR TITLE
Desktop Notifications: subscribe to redux store 

### PR DESCRIPTION
**Pull request by adlk**
This PR is fixing the desktop notification badge and replaces the deprecated `message` event listener with a listener on the redux store – using the newly introduced `hasUnseenNotifications` selector.

While I was at it, I cleaned up the file, removed not longer used code and replaced click events for displaying the notifications panel with the according redux action (`toggleNotificationsPanel`).

#### Test notification badge
- Point calypso submodule to this PR
- [Start app in dev mode ](https://github.com/Automattic/wp-desktop/blob/develop/docs/development.md#dev-mode)
- Create new notification
- Notification badge in dock should appear within a few seconds

#### Test general cleanup
- Navigate through the app with <kbd>Cmd</kbd>+<kbd>1-4</kbd> or via the `Window` menu
- Calypso should navigate to the selected page.

### CODE REVIEW

File: client/lib/desktop/index.js [hunk 1]
```diff
@@ -14,22 +13,23 @@ import { newPost } from 'lib/paths';
...
  * Module variables
  */
-const widgetDomain = 'https://widgets.wp.com';
 
 const Desktop = {
 	/**
 	 * Bootstraps network connection status change handler.
 	 */
-	init: function() {
+	init: async function() {
```
Comment (dmsnell) [inline client/lib/desktop/index.js]: I presume that since this is desktop code that we can use `async` / `await`?

cc: @blowery

Comment (jsnajdr) [reply client/lib/desktop/index.js]: We can (and do) use async/await everywhere in Calypso. Technically, all the Babel transforms are in place. And "officially", we approved the usage on a framework hangout few months ago.

File: client/lib/desktop/index.js [hunk 2]
```diff
@@ -39,42 +39,48 @@ const Desktop = {
...
-
-		location( function( context ) {
-			ipc.send( 'render', context );
-		} );
 	},
 
 	selectedSite: null,
 
+	navigate: function( to ) {
+		if ( isNotificationsOpen( reduxGetState() ) ) {
```
Comment (jsnajdr) [inline client/lib/desktop/index.js]: The store is already in `this.store` and we can call `this.store.getState()`. Only one import from the Redux Bridge module is needed then: `getReduxStore`.

Can it happen that `navigate` is called before the store initialization in the async `init` call is done?

Comment (adlk) [reply client/lib/desktop/index.js]: @jsnajdr oh, good to know! The earliest `navigate` is called is after `this.sendUserLoginStatus` was executed.

**Comment by adlk** [comment]
Merging this based on @jsnajdr feedback in slack.

**Pull request merged**